aips/create_dips_job.py

```diff
@@ -104,22 +99,51 @@ def main(
             session.commit()
         except exc.IntegrityError:
             session.rollback()
-            # POSSIBLE ENHANCEMENT:
-            # Check Aip status and allow retry in some of them
             LOGGER.debug("Skipping AIP (already processed/processing): %s", uuid)
             continue
 
-        create_dip.main(
-            ss_url=ss_url,
-            ss_user=ss_user,
-            ss_api_key=ss_api_key,
+        mets_type = "atom"
+        if args["upload_type"] == "ss-upload":
+            mets_type = "storage-service"
+
+        dip_path = create_dip.main(
+            ss_url=args["ss_url"],
+            ss_user=args["ss_user"],
+            ss_api_key=args["ss_api_key"],
             aip_uuid=uuid,
-            tmp_dir=tmp_dir,
-            output_dir=output_dir,
+            tmp_dir=args["tmp_dir"],
+            output_dir=args["output_dir"],
+            mets_type=mets_type,
         )
 
-        # POSSIBLE ENHANCEMENT:
-        # Save return value from create_dip.main() and update Aip status
+        # Do not try upload on creation error
+        if type(dip_path) == int:
+            LOGGER.error("Could not create DIP from AIP: %s", uuid)
+            continue
```

Comment:
This code here reveals that maybe the "main()" function pattern that we use in automation-tools scripts isn't great particularly in this case where it'd be more convenient/idiomatic to think about exceptions. This is just a thought to be considered for further improvements.